File: Demo/Demo/Shapes/IShape.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Demo
{
	enum Color
	{
		Black,
		White,
		Red,
		Green,
		Blue
	}

	struct Point
	{
		public int X;
		public int Y;

		public Point(int x, int y)
		{
			this.X = x;
			this.Y = y;
		}

		public override string ToString()
		{
			return $"({this.X}, {this.Y})";
		}
	}

	interface IShape
	{
		Color Color { get; set; }
		Point Center { get; }

		void Print(TextWriter output);
	}
}

File: Demo/DemoVisualBasic/Shapes/IShape.vb

Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace Demo.Shapes
	Enum Color
		Black
		White
		Red
		Green
		Blue
	End Enum

	Structure Point
		Public X As Integer
		Public Y As Integer

		Public Sub New(x As Integer, y As Integer)
			Me.X = x
			Me.Y = y
		End Sub

		Public Overrides Function ToString() As String
			Return $"({Me.X}, {Me.Y})"
		End Function
	End Structure

	Interface IShape
		Property Color As Color
		ReadOnly Property Center As Point

		Sub Print(output As TextWriter)
	End Interface
End Namespace
